### PR TITLE
Accept only unique_ptr for distributed vector constructors

### PR DESCRIPTION
**Pull request by MarcelKoch**
Inspired by #1279 I decided to clear up the ownership transfer in the distributed vector constructor. Currently, distributed vectors can be created from existing local vectors by passing them as raw pointers to the constructor. But the constructor will move from the passed in pointer, so this is potentially unsafe. 
One argument in favor of raw pointers was that working with views of local vectors would be simpler (I think?). Since now there is a helper function to create a view of a dense vector, this can now be done as easily with a `unique_ptr` parameter.

To make this work, I had to add `create_const` functions, but I guess that might be useful anyway.

### CODE REVIEW

**Review by upsj — APPROVED** [review]
LGTM! We should probably merge this after the pointer_param PR is merged?

**Comment by MarcelKoch** [comment]
@upsj yes, since this is based on that PR, that makes the most sense.

**Comment by MarcelKoch** [comment]
format-rebase!

**Review by fritzgoebel — APPROVED** [review]
LGTM

**Review by thoasm — CHANGES_REQUESTED** [review]
Tests for new functions `create_const` are missing.

**Review by thoasm — APPROVED** [review]
LGTM!

**Review by upsj — APPROVED** [review]

**Comment by MarcelKoch** [comment]
rebase!

**Comment by MarcelKoch** [comment]
rebase!

**Pull request merged**
